src/pathfinder.rs

```diff
@@ -0,0 +1,283 @@
+use num::integer::sqrt;
+use pathfinding::prelude::astar;
+use serde::{Deserialize, Serialize};
+use std::cell::RefCell;
+use std::hash::Hash;
+use std::num::ParseIntError;
+use std::rc::Rc;
+use thiserror::Error;
+
+thread_local! {
+static NODES: RefCell<Vec<Option<Rc<Node>>>> = RefCell::new(Vec::new());
```

Comment:
Auxmos maint told me to use thread local, which I admit worked pretty well. 

However he told me that I could also do without the Rc since there's already a RefCell. I do not see how it is possible, but I may be missing something